### ISSUE TITLE
Your documentation is really bad.

### ISSUE THREAD

**Issue by TheOGL** [idiotWu/smooth-scrollbar#31]
Hey,

Please explain how to use this plug-in a little better. Your documentation makes no sense. You also have a ton of files in which we have no idea what to do with.

It's a shame because this looks like it could be good.

Thanks,

RS.

**Comment by idiotWu** [comment]
I thought it had been explained clearly enough in wiki pages, I just don't understand what "no sense" means.  You can help me improve it anyway.

**Comment by Yonirt** [comment]
@TheOGL Man, try to do your best, this component and its documentation are excellent, is quite better than other components of the same regard, so don't be a hater and try to be gratefully.

Cheers.

**Comment by TheOGL** [comment]
Sorry if it came off harshly, I was having a bad night trying to find a script, so I just made my own instead and it works great.

Your documentation doesn't explain how to install or use it. Some people don't use NPM or Bower, so then basically you have all these additional files which we have no idea what to do with. I'm a Web Dev with over 15+ years of experience and I'm just lost looking at the docs. While they are obviously advanced instructions, it's always nice to state what things actually are and put things in layman's terms. I tried installing, but I guess I'm missing something.

Cheers,

Ryan

**Comment by TheOGL** [comment]
I'm going to figure out your docs and then rewrite something things for you instead of being a dick about it. lol. ;)

Your scrollbar is very app'esk, so I'm looking forward to using it.

**Comment by neokio** [comment]
@TheOGL has a point. The docs make many assumptions about the developer environment. There is no "all-in-one" basic demo, such as ...

```
<html>
<head>
    <title>basic demo</title>
    <link rel="stylesheet" href="dist/smooth-scrollbar.css">
    <style>
        #myScrollbar {
            max-height: 300px;
        }
    </style>
</head>
<body>

    <section id="myScrollbar" data-scrollbar>
        <img src="xxx.jpg">
    </section>

    <script src="dist/smooth-scrollbar.min.js"></script>
    <script>
        var option = {alwaysShowTracks:true};
        Scrollbar.initAll(options);
    </script>

</body>
```

- There's no quick reference to the size footprint of your script (75k for the JS).
- There's no reference to controlling `height` or `max-height` at all in the docs.
- There's no FAQ / common troubleshooting tips.
- Basically, your docs don't do ANY hand-holding, which will reduce your fanbase by a small but probably significant %.

Despite this, anyone who sees how gorgeously your script performs will take the time to figure it out.

**Comment by idiotWu** [comment]
@neokio 

> There's no quick reference to the size footprint of your script (75k for the JS).

I don't mention it because I think there's no need to worry about content's size today. And the size of outputs depends on the tool you're using, with Babel@5 it's about 60KB and 75KB with Babel@6 :(

> There's no reference to controlling height or max-height at all in the docs.

Well...limited height is necessary for a scrollable area, right? Just like what you do with native scrollbars.

The documentation is bad and I don't know how to improve it, any PR on it will be welcomed :)

**Comment by TheOGL** [comment]
Hey,

Well you could for starters explain how to install the script manually. For instance, what files do we need and what do we add to the the container? Is all we need the .js file? Also, how to we initialize it?

**Comment by TheOGL** [comment]
I'm going through this again because I can't seem to find an all in one scroller.

Looking through the src folder, I see tons of folders with js files... Are we suppose to add all these? Or is there one all in one file?

**Comment by TheOGL** [comment]
I managed to get it working, however does this script support horizontal scrolling? I see that it's supposed to but my mousewheel doesn't move the horizontal bar.

**Comment by idiotWu** [comment]
@TheOGL all right, my 15-year-trained web engineer. You've almost been coding since I was born, so I suppose that you can understand what I'm going to say despite my poor English.

First of all, there's no need to dive into codebase under `src/`, since they are written in ECMAScript 2016+. Have you noticed the big green **[DOWNLOAD](https://github.com/idiotWu/smooth-scrollbar/archive/develop.zip)** button right on the homepage? Download and extract the source code, you can find the compiled code under `dist/` directory. Then:

1. Including stylesheet inside your `<head>` tag: `<link rel="stylesheet" href="dist/smooth-scrollbar.css">`.

2. Importing scripts: `<script src="dist/smooth-scrollbar.js"></script>

3. Writing your scrollable blocks:

    ```html
    <div id="scrollbar">
        some contents...
    </div>
    ```

    Make sure this block is scrollable - a calculatable width&height, a larger content and a smaller container.

4. Initializing smooth-scrollbars in your `app.js`, no need for import declarations as you've imported source files:

    ```javascript
    var scrollbar = Scrollbar.init(document.getElementById('scrollbar');
    ```

5. If you want to play anything with the instance `scrollbar`, check **the API docs in the WiKi page** here: https://github.com/idiotWu/smooth-scrollbar/wiki.

6. Horizontally scroll may be unsupported depends on the type of mouse you are using.

**Comment by TheOGL** [comment]
The first time I looked at it, it was hard to find the script since it was bloated with all separate files. It's also very confusing when you start off by explaining how to install with bower.

I managed to get it all working the other day, however it does not support horizontal scroll, so until it does, it's pretty unusable. You have something good here so I suggest digging into that.

Cheers,

Ryan

**Comment by idiotWu** [comment]
@TheOGL if you can't scroll horizontally now, you might not be able to do it anyway. Because I can't get `deltaX` from your mouse.

So you'd better throw away this shit and find another one or write your own.
